link/freplace.go

```diff
@@ -0,0 +1,155 @@
+package link
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/btf"
+)
+
+type FreplaceLink struct {
+	*RawLink
```

Comment:
Can you please elaborate on what the extra allocation caused by embedding the pointer is?

Reply:
Consistency is good.
I think I was using a pointer to avoid the extra allocation, as the `*RawLink` was already being allocated by the call to `AttachRawLink()`. As far as I can tell, in either case, the compiler is smart enough to optimize things away (at least in the micro-benchmarks I was able to construct to test this type of situation).
Going to change this.

Reply:
Addressed in 80be84a